crates/deltalake-catalog-glue/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+//! Glue Data Catalog.
+//!
+use aws_config::SdkConfig;
+use deltalake_core::data_catalog::{DataCatalog, DataCatalogError};
+use log::*;
+
+#[derive(thiserror::Error, Debug)]
+pub enum GlueError {
+    /// Missing metadata in the catalog
+    #[error("Missing Metadata {metadata} in the Data Catalog ")]
+    MissingMetadata {
+        /// The missing metadata property
+        metadata: String,
+    },
+
+    /// Error calling the AWS SDK
+    #[error("Failed in an AWS SDK call")]
+    AWSError {
+        #[from]
+        source: aws_sdk_glue::Error,
+    },
+}
+
+impl From<GlueError> for DataCatalogError {
+    fn from(val: GlueError) -> Self {
+        DataCatalogError::Generic {
+            catalog: "glue",
+            source: Box::new(val),
+        }
+    }
+}
+
+/// A Glue Data Catalog implement of the `Catalog` trait
+pub struct GlueDataCatalog {
+    client: aws_sdk_glue::Client,
+}
+
+impl GlueDataCatalog {
+    /// Creates a new GlueDataCatalog with environmental configuration
+    pub async fn from_env() -> Result<Self, GlueError> {
+        let config = aws_config::load_from_env().await;
+        let client = aws_sdk_glue::Client::new(&config);
+        Ok(Self { client })
+    }
+
+    /// Create a new [GlueDataCatalog] with the given [aws_config::SdkConfig]
+    pub fn with_config(config: &SdkConfig) -> Self {
+        let client = aws_sdk_glue::Client::new(config);
+        Self { client }
+    }
+}
+
+impl std::fmt::Debug for GlueDataCatalog {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(fmt, "GlueDataCatalog")
+    }
+}
+
+// Placeholder suffix created by Spark in the Glue Data Catalog Location
+const PLACEHOLDER_SUFFIX: &str = "-__PLACEHOLDER__";
```

Comment:
```suggestion
const PLACEHOLDER_SUFFIX: &str = "__PLACEHOLDER__";
```

Reply:
... or rather looking at the name "suffix" this may actually be intended, but making sure nonetheless 😆.